src/lib/__tests__/routing/external-influence-test.ts

```diff
@@ -0,0 +1,73 @@
+import { createSearchClient } from '../../../../test/mock/createSearchClient';
+import { wait } from '../../../../test/utils/wait';
+import historyRouter from '../../routers/history';
+import instantsearch from '../../..';
+import { connectSearchBox } from '../../../connectors';
+
+const writeDelay = 10;
+const writeWait = 1.5 * writeDelay;
+
+describe('routing', () => {
+  beforeEach(() => {
+    window.history.replaceState({}, '', 'http://localhost/');
+    jest.clearAllMocks();
+  });
+
+  test('Something else updates the URL: IS.js stay responsive', async () => {
```

Comment:
**question:** The only difference I see between this test and the one in `src/lib/__tests__/routing/spa-test.ts` is that there's no intermediary `dispose()`. What use case does this represent? We might need a comment (and/or change the test description) to make it clear for later reading.

Reply:
this is for when InstantSearch does not dispose, but something else calls pushState in the middle, without navigating off the InstantSearch page